pkg/libvmi/accesscredentials.go

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors
+ *
+ */
+
+package libvmi
+
+import (
+	v1 "kubevirt.io/api/core/v1"
+)
+
+// WithSSHPublicKey adds an AccessCredential that propagates the public keys found
+// in secretName to the authorized_keys file of the user with name userName via the
+// qemu-guest-agent.
+func WithSSHPublicKey(secretName, userName string) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		if vmi.Spec.AccessCredentials == nil {
+			vmi.Spec.AccessCredentials = []v1.AccessCredential{}
+		}
+		vmi.Spec.AccessCredentials = append(vmi.Spec.AccessCredentials, v1.AccessCredential{
+			SSHPublicKey: &v1.SSHPublicKeyAccessCredential{
+				Source: v1.SSHPublicKeyAccessCredentialSource{
+					Secret: &v1.AccessCredentialSecretSource{
+						SecretName: secretName,
+					},
+				},
+				PropagationMethod: v1.SSHPublicKeyAccessCredentialPropagationMethod{
+					QemuGuestAgent: &v1.QemuGuestAgentSSHPublicKeyAccessCredentialPropagation{
+						Users: []string{userName},
+					},
+				},
+			},
+		})
+	}
+}
+
+// WithUserPassword adds an AccessCredential that propagates the user passwords found
+// in secretName via the qemu-guest-agent.
+func WithUserPassword(secretName string) Option {
```

Comment:
The name here may collide with some other user-password usage.

How about you prefix it with `AccessCredential`: `WithAccessCredentialUserPassword`?
Or you can use a shortcut `AccCred`...
I'm not very strongly opinionated about this, but I do see a potential of naming collision.

If you do it, then it should be done in the other builder as well I guess.

Reply:
Updated to add the prefix.